java/org/openobservatory/measurement_kit/nettests/TestListener.java

```diff
@@ -0,0 +1,114 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+package org.openobservatory.measurement_kit.nettests;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.support.v4.content.LocalBroadcastManager;
+
+import org.openobservatory.measurement_kit.common.LogCallback;
+
+public class TestListener {
+
+    private LocalBroadcastManager lbm;
+    private Integer testId;
+
+    // When we register a BroadcastReveiver we need to unregister it
+    // when we don't need it more (like when we close the activity).
+    // To handle this case in a safer way I save the receiver created
+    // without giving it back to the caller, so it's my responsibility
+    // to keep track of it, but it remain the responsibility of the
+    // developer to call clear to unregister the listener
```

Comment:
I think this comment should be improved, because, as is, I don't fully understand it.